core/services/ocr3/plugins/ccip/commit/plugin_spec.py

```diff
@@ -0,0 +1,128 @@
+#
+# High-level Python specification for the CCIP OCR3 Commit Plugin.
+#
+# This specification aims to provide a clear and comprehensive understanding
+# of the plugin's functionality. It is highly recommended for engineers working on CCIP
+# to familiarize themselves with this specification prior to reading the
+# corresponding Go implementation.
+#
+class CommitPlugin:
+    def __init__(self):
+         self.cfg = {
+             "dest_chain": "chainD",
+             "f_chain": {"chainA": 2, "chainB": 3, "chainD": 3},
+             "observer_info": {
+                 "nodeA": {
+                     "supported_chains": {"chainA", "chainB", "chainD"},
+                     "token_prices_observer": True,
+                 }
+             },
+             "priced_tokens": {"tokenA", "tokenB"},
+         }
+         self.keep_cfg_in_sync()
+
+    def query(self):
+        pass
+
+    def observation(self, previous_outcome):
+        # Observe last msg sequence numbers for each source chain: {sourceChain: sequenceNumber}
+        observed_seq_nums = previous_outcome.get("observed_seq_nums", {})
+        if self.can_read_dest():
+            on_chain_seq_nums = self.offRamp.get_sequence_numbers()
+            for (chain, seq_num) in on_chain_seq_nums.items():
+                if chain not in observed_seq_nums or seq_num > observed_seq_nums[chain]:
+                    observed_seq_nums[chain] = seq_num
+
+        # Observe new msgs: {sourceChain: [(id, seq_num)]}
+        new_msgs = {}
+        for (chain, seq_num) in observed_seq_nums.items():
+            if self.can_read(chain):
+                new_msgs[chain] = self.onRamp(chain).get_msgs(chain, start=seq_num+1, limit=256)
+                for msg in new_msgs[chain]:
+                    assert(msg.id == msg.compute_id())
+
+        # Observe token prices. {token: price}
+        token_prices = self.get_token_prices()
+
+        # Observe gas prices. {chain: gasPrice}
+        gas_prices = self.get_gas_prices()
+
+        # Observe fChain for each chain
```

Comment:
```suggestion
        # Observe fChain for each chain {chain: f_chain}
```

Similar to the other comments